### PR TITLE
write.mapie flexible for time .nc variable naming

### PR DESCRIPTION
**Pull request by emolinab**
change in write.magpie to make more flexible the variables name in the generated .nc maps

### CODE REVIEW

File: R/write.magpie.R [hunk 1]
```diff
@@ -92,7 +93,7 @@
 #' @importFrom utils write.csv write.table
 #' @export
 write.magpie <- function(x, file_name, file_folder = "", file_type = NULL, append = FALSE, comment = NULL, # nolint
-                         comment.char = "*", mode = NULL, ...) {       # nolint
+                         comment.char = "*", mode = NULL, zname = NULL,...) {       # nolint
```
Comment (tscheypidi) [inline R/write.magpie.R:96]:
```suggestion
                         comment.char = "*", mode = NULL, zname = "Time", ...) {       # nolint
```

File: R/write.magpie.R [hunk 2]
```diff
@@ -196,9 +197,11 @@ write.magpie <- function(x, file_name, file_folder = "", file_type = NULL, appen
         x <- x[[1]]
       }
       if (is.null(varname)) varname <- "Variable"
+      zname <- if(!is.null(zname)) zname else "Time"
```
Comment (tscheypidi) [inline R/write.magpie.R:200]: I think this line is not needed if you set `zname = "Time"` by default

File: R/write.magpie.R [hunk 3]
```diff
@@ -196,9 +197,11 @@ write.magpie <- function(x, file_name, file_folder = "", file_type = NULL, appen
...
       raster::writeRaster(x, filename = filePath, format = format[file_type], overwrite = TRUE,
-                          zname = "Time", zunit = zunit, varname = varname, ...)
+                          zname = zname, zunit = zunit, varname = varname, ...)
     } else if (file_type == "nc") {
+      zname <- if(!is.null(zname)) zname else "Time"
```
Comment (tscheypidi) [inline R/write.magpie.R:204]: I think this line is not needed if you set `zname = "Time"` by default

**Review by tscheypidi — CHANGES_REQUESTED** [review]
I would suggest to set `zname = "Time"` by default for reduce a bit the changes in the code and to make it more transparent how `zname` will be set.

In addition, the lucode2 check complains about the following:

```
❯ checking R code for possible problems ... NOTE
  Found if() conditions comparing class() to string:
  File ‘magclass/R/write.magpie.R’: if (class(x) == "RasterBrick") ...
  File ‘magclass/R/write.magpie.R’: if (class(x) == "RasterBrick") ...
  ```
  This is unrelated to your changes and just shows up now due to more strict tests from R, but it would be nice if you could fix this as well as part of your PR. The fix is rather easy: Just replace  ‘if (class(x) == "RasterBrick")` by `if (inherits(x, "RasterBrick"))`

File: R/write.magpie.R (hunk 3 above)
Comment (emolinab) [reply R/write.magpie.R:204]: Okie dokie, makes sense!

**Pull request merged**